test/alternative_spec.rb

```diff
@@ -0,0 +1,51 @@
+require_relative "spec_helper.rb"
+
+describe "UpdateAlternatives::Alternative" do
+
+  context "when we get all alternatives names" do
+    it "should return an array with all alternatives names" do
+
+      alternatives_names_stub
+      expected_names = %w(pilconvert pip rake rake.ruby2.1 rdoc rdoc.ruby2.1 ri ri.ruby2.1)
+
+      alternatives_name_list = UpdateAlternatives::Alternative.all_names
+
+      expect(alternatives_name_list).to eq expected_names
+    end
+  end
+
+  describe ".all" do
+    subject { UpdateAlternatives::Alternative.all }
+
+    it "returns an array of Alternative objects" do
+      alternatives_pip_with_two_choices_stub
+      expect(subject.class).to eq Array
+      expect(subject.all? { |e| e.is_a?(UpdateAlternatives::Alternative) }).to eq true
```

Comment:
just try to make this expect failed...it return quite useless output `false is not true`. Better way is to have own expect for all objects like `subject.all? { |e| expect(e).to be_a(UpdateAlternatives::Alternative) }` which produce much nicer output when failed like `String is not a UpdateAlternatives::Alternative` with probably more details about given string.

When you write test always try to help future maintainer. So when test start failing, it should produce helpful output as for YaST it can happen that test failed only on alien hardware located on Mars, which is hard to reach, so having all info in output is very helpful.